src/main/java/org/opensearch/ad/model/AnomalyResult.java

```diff
@@ -89,42 +98,107 @@
-            dataStartTime,
-            dataEndTime,
-            executionStartTime,
-            executionEndTime,
-            error,
-            null,
-            user,
-            schemaVersion
-        );
-    }
+    // sequence index (the number of updates to RCF) -- it is possible in imputation
+    // that
+    // the number of updates more than the input tuples seen by the overall program
+    private final Long totalUpdates;
+
+    // flag indicating if the anomaly is the start of an anomaly or part of a run of
+    // anomalies
+    private final Boolean startOfAnomaly;
+
+    // flag indicating if the time stamp is in elevated score region to be
+    // considered as anomaly
+    private final Boolean inHighScoreRegion;
+
+    /**
+     * position of the anomaly vis a vis the current time (can be -ve) if anomaly is
+     * detected late, which can and should happen sometime; for shingle size 1; this
+     * is always 0
+     */
+    private final Integer relativeIndex;
+
+    // a flattened version denoting the basic contribution of each input variable
+    private final double[] currentTimeAttribution;
 
+    /*
+    oldValues is related to relativeIndex and startOfAnomaly.
+    So if we detect anomaly late, we get the baseDimension values from the past (current is 0).
+    That is, we look back relativeIndex * baseDimensions.
+    
+    For example, current shingle is
+    "currentValues": [
+    6819.0,
+    2375.3333333333335,
+    0.0,
+    49882.0,
+    92070.0,
+    5084.0,
+    2072.809523809524,
+    0.0,
+    43529.0,
+    91169.0,
+    8129.0,
+    2582.892857142857,
+    12.0,
+    54241.0,
+    84596.0,
+    11174.0,
+    3092.9761904761904,
+    24.0,
+    64952.0,
+    78024.0,
+    14220.0,
+    3603.059523809524,
+    37.0,
+    75664.0,
+    71451.0,
+    17265.0,
+    4113.142857142857,
+    49.0,
+    86376.0,
+    64878.0,
+    16478.0,
+    3761.4166666666665,
+    37.0,
+    78990.0,
+    70057.0,
+    15691.0,
+    3409.690476190476,
+    24.0,
+    71604.0,
+    75236.0
+    ],
+    Since relativeIndex is -2, we look back baseDimension * 2 and get the oldValues:
+    "startOfAnomaly": true,
+    "inHighScoreRegion": true,
+    "relativeIndex": -2,
+    "oldValues": [
+    17265.0,
+    4113.142857142857,
+    49.0,
+    86376.0,
+    64878.0
+    ],
+    
+    So oldValues is null when relativeIndex is 0 or startOfAnomaly is true.*/
+    private final double[] oldValues;
+
+    // expected values, currently set to maximum 1 expected. In the future, we
+    // might give different expected values with differently likelihood. So
+    // the two-dimensional array allows us to future-proof our applications.
+    // Also, expected values correspond to oldValues if present or current input
+    // point otherwise. If oldValues is present, it will take effort to show this
+    // on UX since we found an anomaly from the past (old values).
+    private final double[][] expectedValuesList;
```

Comment:
How about add some examples to help understand what the 2D array will looks like and what each dimension means?